### ISSUE TITLE
Add inverse of waitForSelector visible: true (waitForNotSelector?)

### ISSUE THREAD

**Issue by quicksnap** [puppeteer/puppeteer#549]
It would be very useful to test when a selector is no longer visible. For instance, I'm waiting for a certain container not to have a `loading` class and then assert things.

Would you accept a PR for this? It wouldn't be tough to implement. If so, what should the API look like?

As it stands now, it's necessary to do some dance with `waitForFunction`.

**Comment by aslushnikov** [comment]
> As it stands now, it's necessary to do some dance with waitForFunction.

If I understand correctly, this behavior is easily implementable with `waitForFunction`:

```js
// wait for ".box" to disappear
await page.waitForFunction(() => !document.querySelector('.box'), {polling: 'mutation'});
```

**Comment by quicksnap** [comment]
@aslushnikov Yeah, I did something similar. I didn't even need to specify the polling option and it worked just fine.

Still, seems like it would be a very useful utility. If it's worth adding to the API, just needs bikeshedding.

**Comment by aslushnikov** [comment]
Closing this for now. We'll add this to the API if there are many clients who demand this and who are not satisfied with the one-liner for some reason.

**Comment by aslushnikov** [comment]
Closed with PR #967
